implicits/src/com/rallyhealth/weepickle/v1/implicits/internal/Macros.scala

```diff
@@ -390,9 +390,17 @@ object Macros {
         """
 
         /**
-          * @see [[shouldDropDefault()]]
+          * Transforming the argument to itself applies any logic that the client may have provided
+          * in the implicit definition of From and/or To for the data type. This is important as this
+          * logic may transform the value into a default value which needs to be dropped.
+          *
+          * @see [[Argument.omitDefault]].
           */
-        if (arg.writingCheckDefault) q"""if (v.${TermName(arg.raw)} != ${arg.default}) $snippet"""
+        if (arg.writingCheckDefault)
+          q"""
+             val fromToArg = implicitly[${c.prefix}.FromTo[${arg.argType}]]
+             if (fromToArg.transform(v.${TermName(arg.raw)}, fromToArg) != ${arg.default}) $snippet
```

Comment:
There's another check to calculate the length on :413 that needs to be consistent with this, or else I'd expect the MessagePack support to be broken.

Reply:
If those checks are covered in the unit tests, then it works. If they aren't, then we are missing tests.

Reply:
P.S. Philosophically speaking, if there are missing tests, it is just as likely that this change fixes a `MessagePack` bug as introduces one.

Reply:
Fair. I'll volunteer to add tests for those scenarios if we get past the other concerns and want to continue with this approach.